src/PolyType.Roslyn/KnownSymbols.cs

```diff
@@ -19,6 +24,18 @@ public class KnownSymbols(Compilation compilation)
     public IAssemblySymbol CoreLibAssembly => _CoreLibAssembly ??= Compilation.GetSpecialType(SpecialType.System_Int32).ContainingAssembly;
     private IAssemblySymbol? _CoreLibAssembly;
 
+    /// <summary>
+    /// Gets the symbol for TypeShapeExtensionAttribute.
+    /// </summary>
+    public INamedTypeSymbol TypeShapeExtensionAttribute => _TypeShapeExtensionAttribute ??= Compilation.GetTypeByMetadataName("PolyType.TypeShapeExtensionAttribute") ?? throw new InvalidOperationException("TypeShapeExtensionAttribute not found.");
```

Comment:
I generally try to keep all PolyType runtime attributes in the `PolyType.SourceGenerator` project and not in `PolyType.Roslyn` since the latter is supposed to be a library-agnostic type graph traversal component.

Reply:
I'll move it.